src/plugins/data/common/es_query/kuery/functions/nested.js

```diff
@@ -28,10 +28,6 @@ export function buildNodeParams(path, child) {
 }
 
 export function toElasticsearchQuery(node, indexPattern, config, context = {}) {
-  if (!indexPattern) {
-    throw new Error('Cannot use nested queries without an index pattern');
-  }
-
   const [path, child] = node.arguments;
   const stringPath = ast.toElasticsearchQuery(path);
```

Comment:
Seems `indexPattern` in `ast.toElasticsearchQuery` is no required

![image](https://user-images.githubusercontent.com/9773803/70155927-648ada80-16b3-11ea-9ff8-28b143d27d82.png)